pandas/tests/indexes/datetimes/test_datetime.py

```diff
@@ -236,7 +236,8 @@ def test_stringified_slice_with_tz(self):
         start = datetime.datetime.now()
         idx = DatetimeIndex(start=start, freq="1d", periods=10)
         df = DataFrame(lrange(10), index=idx)
-        df["2013-01-14 23:44:34.437768-05:00":]  # no exception here
+        with tm.assert_produces_warning(UserWarning):
+            df["2013-01-14 23:44:34.437768-05:00":]  # no exception here
```

Comment:
I would remove the ``# no exception here`` and add a comment about the warning that is produced

Reply:
I am not sure whether that is a good idea. That comment is not mine and it is not related to my code. The proposed refactoring is beyond the scope of this pull request.